src/api/app/components/comment_bubble_component.html.haml

```diff
@@ -0,0 +1,63 @@
+.comment-bubble.ms-3
+  - if policy(comment).update? || policy(comment).destroy?
+    .dropdown.dropleft.float-end.mt-3.mx-3
+      = link_to('#', role: 'button', 'data-bs-toggle': 'dropdown', 'aria-expanded': 'false') do
+        %i.fas.fa-ellipsis
+      .dropdown-menu
+        - if policy(comment).update?
+          = link_to("#edit_form_of_#{comment.id}", id: "edit_button_of_#{comment.id}", 'data-bs-toggle': 'collapse',
+                    class: 'dropdown-item collapsed') do
+            Edit
+        - if Flipper.enabled?(:content_moderation, User.session)
+          - if policy(comment).moderate?
+            = button_to(comment.moderated? ? 'Permit' : 'Moderate', moderate_comment_path(comment),
+                        class: 'dropdown-item', remote: true, form_class: 'moderate-form', params: { moderation_state: !comment.moderated? })
+        - if policy(comment).destroy?
+          = link_to('#', data: { 'bs-toggle': 'modal',
+                                'bs-target': '#delete-comment-modal',
+                                confirmation_text: 'Please confirm deletion of comment',
+                                action: comment_path(comment) },
+                        class: 'dropdown-item delete_link', title: 'Delete comment') do
+            Delete
+  .px-3.py-2.border-bottom
+    - unless hideusername
```

Comment:
I think this can be solved in a more elegant manner with [template variants](https://viewcomponent.org/guide/templates.html#templates)